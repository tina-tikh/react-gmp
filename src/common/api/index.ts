import * as MovieService from './movie-service';

export { MovieService };
