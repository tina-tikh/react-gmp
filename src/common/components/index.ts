import ErrorBoundary from './ErrorBoundary';
import FilmList from './FilmList';

export * from './form';
export * from './layout';
export { ErrorBoundary, FilmList };
