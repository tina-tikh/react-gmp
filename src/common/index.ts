export * from './api';
export * from './components';
export * from './models';
export * from './theme';
