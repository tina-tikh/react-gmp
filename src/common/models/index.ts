import Movie from './movie';
import MoviesQueryResponse from './moviesQueryResponse';

export { Movie, MoviesQueryResponse };
