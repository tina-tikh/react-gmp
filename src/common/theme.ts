export type Theme = {
    colors: {
        primary: string,
        secondary: string,
        light: string,
        dark: string,
        darker: string,
        darkest: string
    }
};

const theme: Theme = {
    colors: {
        primary: '#f65261',
        secondary: '#a1e66f',
        light: '#fff',
        dark: '#555',
        darker: '#424242',
        darkest: '#232323'
    },
};

export default theme;
