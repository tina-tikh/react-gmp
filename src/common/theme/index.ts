import theme from './theme';
import styled from './styled';

export { styled, theme };
