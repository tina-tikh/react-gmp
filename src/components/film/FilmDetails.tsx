import * as React from 'react';
import { Component, ReactNode } from 'react';

import { styled } from '../../theme';
import { Movie } from '../../store/types';

type FilmDetailsProps = {
  film: Movie;
};

const Poster = styled.img`
  float: left;
  height: 40rem;
  margin-right: 2rem;
`;

const Title = styled.h4`
  margin: 2rem;
  font-size: 3rem;
`;

const Vote = styled.span`
  display: inline-block;
  width: 5rem;
  height: 5rem;
  margin: 1rem;
  border-radius: 5rem;
  text-align: center;
  line-height: 5.2rem;
  font-size: 2.5rem;
  font-weight: normal;
  color: ${(props) => props.theme.colors.secondary};
  border: 1px solid ${(props) => props.theme.colors.light};
`;

const Info = styled.div`
  margin: 2rem;
  font-size: 2rem;
  color: ${(props) => props.theme.colors.primary};

  span + span {
    margin-left: 2rem;
  }
`;

const Overview = styled.div`
  margin: 2rem;
  font-size: 2rem;
`;

class FilmDetails extends Component<FilmDetailsProps> {
  render(): ReactNode {
    return (
      <div>
        <Poster src={this.props.film?.poster_path} />
        <div>
          <Title>
            {this.props.film?.title}
            <Vote>{this.props.film?.vote_average}</Vote>
          </Title>
          <Info>
            <span>{this.getYear(this.props.film?.release_date)}</span>
            <span>{this.props.film?.runtime}</span>
          </Info>
          <Overview>{this.props.film?.overview}</Overview>
        </div>
      </div>
    );
  }

  getYear(date: string): number {
    return date && new Date(date).getFullYear();
  }
}

export default FilmDetails;
