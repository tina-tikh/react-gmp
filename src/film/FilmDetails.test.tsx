import * as React from 'react';
import { mount } from 'enzyme';

import { Movie, Page } from '../common';
import FilmDetails from './FilmDetails';

describe("<FilmDetails />", () => {
  const film: Movie = {
    "id": 447365,
    "title": "Guardians of the Galaxy Vol. 3",
    "tagline": "",
    "vote_average": 0,
    "vote_count": 9,
    "release_date": "2020-05-01",
    "poster_path": "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
    "overview": "The third film based on Marvel's Guardians of the Galaxy.",
    "budget": 0,
    "revenue": 0,
    "genres": ["Action", "Adventure", "Science Fiction"],
    "runtime": null
  };

  it("should be rendered", () => {
    const component = mount(<Page><FilmDetails film={film}/></Page>);
    expect(component).toMatchSnapshot();
  });
});
