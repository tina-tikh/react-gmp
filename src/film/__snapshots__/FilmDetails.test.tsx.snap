// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmDetails /> should be rendered 1`] = `
<div>
  <Styled(img)
    src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
  />
  <div>
    <Styled(h4)>
      Guardians of the Galaxy Vol. 3
      <Styled(span)>
        0
      </Styled(span)>
    </Styled(h4)>
    <Styled(div)>
      <span>
        2020
      </span>
      <span />
    </Styled(div)>
    <Styled(div)>
      The third film based on Marvel's Guardians of the Galaxy.
    </Styled(div)>
  </div>
</div>
`;
